webapp/src/features/intersections/reports/report-list-filters.tsx

```diff
@@ -0,0 +1,173 @@
+  onChange: (filters: ReportListFilter) => void
+  onClose: () => void
+  open: boolean
+  loading: boolean
+  containerRef: any
+  setOpenReportGenerationDialog: (open: boolean) => void
+}
+
+export const ReportListFilters = (props: ReportListFiltersProps) => {
+  const { containerRef, filters, onChange, onClose, open, loading, setOpenReportGenerationDialog, ...other } = props
+  const [currentFilters, setCurrentFilters] = useState(filters)
+  const [filtersValid, setFiltersValid] = useState([true, ''])
+
+  useEffect(() => {
+    updateFiltersValid()
+  }, [currentFilters])
+
+  const startDateChange = (date: Date | null) => {
+    setCurrentFilters({ ...currentFilters, startDate: date })
+  }
+
+  const endDateChange = (date: Date | null) => {
+    setCurrentFilters({ ...currentFilters, endDate: date })
+  }
+
+  const updateFilters = () => {
+    onChange?.(currentFilters)
+  }
+
+  const updateFiltersValid = () => {
+    let filtersValidLocal = true
+    const reasons: string[] = []
+    if (currentFilters.startDate === null || currentFilters.endDate === null) {
+      filtersValidLocal = false
+      reasons.push('Start Date and End Date must be set')
+    } else if (currentFilters.startDate >= currentFilters.endDate) {
+      filtersValidLocal = false
+      reasons.push('Start Date must be before End Date')
+    }
+    // TODO: validate dates
+    setFiltersValid([filtersValidLocal, reasons.join('\n')])
+    if (filtersValidLocal) {
+      onChange?.(currentFilters)
+    }
+  }
+
+  const content = (
+    <Box
+      sx={{
+        pb: 1,
+        pt: 1,
+        px: 3,
+      }}
+    >
+      <Box
+        sx={{
+          mb: 2,
+        }}
+      >
+        <IconButton onClick={onClose}>
+          <Close fontSize="small" />
+        </IconButton>
+      </Box>
+      <Typography color="textSecondary" sx={{ mt: 3 }} variant="subtitle2">
+        Issue date
+      </Typography>
+      <Stack spacing={2} sx={{ mt: 2 }}>
+        <LocalizationProvider dateAdapter={AdapterDateFns}>
+          <DateTimePicker
+            label="From"
+            onChange={startDateChange}
+            renderInput={(inputProps) => <TextField {...inputProps} />}
+            value={currentFilters.startDate}
+            disabled={loading}
+          />
+          <DateTimePicker
+            label="To"
+            onChange={endDateChange}
+            renderInput={(inputProps) => <TextField {...inputProps} />}
+            value={currentFilters.endDate}
+            disabled={loading}
+          />
+        </LocalizationProvider>
+      </Stack>
+      {!filtersValid[0] && (
+        <Typography color="red" sx={{ mt: 3 }} variant="subtitle2">
+          Invalid filters: {filtersValid[1]}
+        </Typography>
+      )}
+      <Box sx={{ m: 1, position: 'relative', display: 'flex', alignItems: 'center' }}>
+        {loading && (
+          <CircularProgress
+            size={24}
+            sx={{
+              // color: green[500],
```

Comment:
Is this work keeping since it seems to look fine?

Reply:
removed